### DISCUSSION TITLE
Symfony Mailer doesn't work on prod server

### DISCUSSION THREAD

**Discussion by Stefan39** [symfony/symfony#44795] [Q&A]
Merry Christmas everyone,

i try to use Symfony Mailer (5.3) on my Web Host Hetzner, where the port 465 is blocked. So i must use the port 587 and i have try it with Sendgrid DSN like
`MAILER_DSN=sendgrid+smtp://API_KEY@smtp.sendgrid.net:587`

and also i have trying from gmail:
`MAILER_DSN=gmail+smtp://test@gmail.com:password@smtp.gmail.com:587?verify_peer=0`

(both also without +smtp, but nothing changed)

Both one produces an Error with the following message:
`[2021-12-25T09:43:13.271464+01:00] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Messenger\Exception\HandlerFailedException: "Handling "Symfony\Component\Mailer\Messenger\SendEmailMessage" failed: Connection could not be established with host "ssl://smtp.gmail.com:465": stream_socket_client(): Unable to connect to ssl://smtp.gmail.com:465 (Connection refused)" at /usr/www/users/[username]/[domain]/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php line 80 {"exception":"[object] (Symfony\\Component\\Messenger\\Exception\\HandlerFailedException(code: 0): Handling \"Symfony\\Component\\Mailer\\Messenger\\SendEmailMessage\" failed: Connection could not be established with host \"ssl://smtp.gmail.com:465\": stream_socket_client(): Unable to connect to ssl://smtp.gmail.com:465 (Connection refused) at /usr/www/users/[username]/[domain]/vendor/symfony/messenger/Middleware/HandleMessageMiddleware.php:80)\n[previous exception] [object] (Symfony\\Component\\Mailer\\Exception\\TransportException(code: 0): Connection could not be established with host \"ssl://smtp.gmail.com:465\": stream_socket_client(): Unable to connect to ssl://smtp.gmail.com:465 (Connection refused) at /usr/www/users/[username]/[domain]/vendor/symfony/mailer/Transport/Smtp/Stream/SocketStream.php:139)"} []`

I have also try to use sendgrid+api like:
`MAILER_DSN=sendgrid+api://API_KEY@default`

with produce the error:

`request.CRITICAL: Uncaught PHP Exception LogicException: "You cannot use "Symfony\Component\Mailer\Transport\AbstractHttpTransport" as the HttpClient component is not installed. Try running "composer require symfony/http-client"." at /usr/www/users/[username]/[domain]/vendor/symfony/mailer/Transport/AbstractHttpTransport.php line 36 {"exception":"[object] (LogicException(code: 0): You cannot use \"Symfony\\Component\\Mailer\\Transport\\AbstractHttpTransport\" as the HttpClient component is not installed. Try running \"composer require symfony/http-client\". at /usr/www/users/[username]/[domain]/vendor/symfony/mailer/Transport/AbstractHttpTransport.php:36)"} []`

but the Http_Client (5.3.x) is installed via composer.

I have also try to clear the cache various times:
`APP_ENV=prod APP_DEBUG=0 php bin/console clear:cache`

and also delete the /var/cache directory directly on the terminal, nothing helps (always the same error).

I have also open a support ticket on sendgrid, but they have also no idea what's going wrong here. I'm also know that exists a bug in github (see https://github.com/symfony/symfony/issues/36224) but I couldn't believe that nobody have this problem in the past.

How i can use now the mailer now btw. how i can change the using port?

**Comment by plejus** [comment]
I have the same problem, the same configuration works on localhost docker:

`Handling "Symfony\Component\Mailer\Messenger\SendEmailMessage" failed: Connection could not be established with host "ssl://smtp.sendgrid.net:465": stream_socket_client(): Unable to connect to ssl://smtp.sendgrid.net:465 (Connection refused)
`

config:
`MAILER_DSN=sendgrid://API_KEY@default`

**Reply by grizzlii** [reply]
Same problem here. I was using Swift Mailer, which is supposedly now deprecated and have tested Symfony Mailer and PHPMailer. Swift and PHPMailer work flawlessly connecting to my mail host. I have tried a hundred ways to connect with Symfony Mailer. It just doesn't work. Always get this error: TransportException: Connection could not be established with host and a bunch on unintelligible error code. This code is absurdly complicated, their docs are virtually non-existent, there are no working examples or sample code and the software is buggy as hell. Not to mention IT DOESN'T WORK! Don't waste yer time!

**Reply by plejus** [reply]
@grizzlii I found workaround for SendGrid (my case) - I used API transport instead of SMTP, configuration look like:
`MAILER_DSN=sendgrid+api://SG.{{api_key_here}}@default`

**Reply by grizzlii** [reply]
That's great but I am not using SendGrid. This is simply secure SMTP to my mail hosting account. All I need is to connect to my account with username, pword, protocol and port. I don't need all the other kookie crap in Sym Mailer. Not only does this interface not accept 'real' secure passwords, it cannot connect to my mail server with any combination of real world settings. This should be really simple stuff. My settings work fine in Swift and PHPMailer, which is what I will use from now on, but not with Sym Mailer and no clues. For the time I have spent screwing around with this thing I could have written my own app from scratch. The benefit of frameworks is to encapsulate complex processes and provide an efficient API that save time and provides a robust solution. If that is what you want, run for the hills. This ain't it!

**Reply by FrancoisFarlan** [reply]
Hello, I encounter the same problem with SendinBlue. Has anyone found a fix for this ?

**Reply by eerison** [reply]
Hey @grizzlii have you solved your issue?

> This is simply secure SMTP to my mail hosting account. All I need is to connect to my account with username, pword, protocol and port. I don't need all the other kookie crap in Sym Mailer. 

as I can see you are trying to use a simple SMTP, Have you tried to use https://symfony.com/doc/current/mailer.html#transport-setup?

```bash
MAILER_DSN=smtp://user:pass@smtp.example.com:port
```

**Comment by eerison** [comment]
Hey @Stefan39 

as I can see your issue was solved here: https://github.com/symfony/symfony/issues/36224, wasn't it?

**Comment by grizzlii** [comment]
Yes I solved this problem. Using PHPMailer instead.

**Reply by FlairmAmaan** [reply]
how ?

**Reply by FlairmAmaan** [reply]
can u please help i am getting error on using sendmail in laravel on server but works fine on localhost

error- 
Symfony
 \ 
Component
 \ 
Mailer
 \ 
Exception
 \ 
TransportException
Connection could not be established with host "smtp.gmail.com:587": stream_socket_client(): php_network_getaddresses: getaddrinfo for smtp.gmail.com failed: Temporary failure in name resolution

**Reply by eerison** [reply]
if it is working locally, probably the server isn't resolving DNS.

**Reply by FlairmAmaan** [reply]
do you have any suggestion that how can i fix server DNS ?

**Reply by eerison** [reply]
I don't know how it was configured on your server, But if you have access to it, you should ping for any dns name, like

```
ping smtp.gmail.com
PING smtp.gmail.com (142.250.27.109): 56 data bytes
64 bytes from 142.250.27.109: icmp_seq=0 ttl=107 time=24.426 ms
64 bytes from 142.250.27.109: icmp_seq=1 ttl=107 time=28.004 ms
64 bytes from 142.250.27.109: icmp_seq=2 ttl=107 time=27.881 ms
```

if it doesn't work on the server, just such on google how to `resolve dns name`, there are a lot of articles about it :)

I hope it helps you :)

**Reply by FlairmAmaan** [reply]
sure i will , thanks for ur help :)

**Comment by eerison** [comment]
Well in theory it should work for you 

```
MAILER_DSN=smtp://user:pass@smtp.example.com:port
```
